### DISCUSSION TITLE
How to connect multiple Edge from the same Node?

### DISCUSSION THREAD

**Discussion by li-jia-nan** [xyflow/xyflow#3969] [Q&A]
How to connect multiple Edge from the same Node ? like this:

<img width="989" alt="image" src="https://github.com/xyflow/xyflow/assets/49217418/02b2b408-e34e-4b81-8ebc-c9eeceabaff5">

**Comment by bcakmakoglu** [comment]
Check the [`Custom Node`](https://reactflow.dev/examples/nodes/custom-node) example.
It demonstrates how to add *multiple handles* to a node (which, from the image you've shown, is what you're looking for).

When using multiple handles, make sure each handle has a unique id `<Handle id={handleId} ...>` as you will need that id to describe from what handle your edges start / end, for example

```js
{
	id: 'e1-2',
	source: '1',
	sourceHandle: '1a',
	target: '2',
	targetHandle: '2a'
}
```

To align handles to a specific element (as in your screenshot) make sure that the wrapping element has `position: relative` on it.
`<Handle>` components use `position: absolute`, meaning they will align to whatever wrapping element closest to it has `position: relative`

```jsx
<div style={{ position: 'relative' }}>
	<Handle ...>
```
